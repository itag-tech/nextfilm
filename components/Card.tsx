import { FC } from "react"
import clsx from "clsx"
import Image from "next/image"

import { Movie } from "../models/Movie"
import placeholderImage from "../public/placeholder-film.png"

type CardProps = {
  movie: Movie
}

const Card: FC<CardProps> = ({ movie }) => {

  const poster = (movie.Poster === "N/A" && placeholderImage) || movie.Poster

  return (
    <div className={clsx("flex flex-col justify-center items-center bg-white bg-opacity-50 rounded-lg shadow-md truncate h-64 w-80")}>
      <Image
        src={poster}
        alt={movie.Title}
        width={200}
        height={200}
        className={clsx("h-32 w-32 object-cover shadow-lg")}
      />
      <div className={clsx("flex flex-col justify-center items-center")}>
        <p className={clsx("text-xl font-medium truncate max-w-xs text-purple-700 m-4 px-6")}>{movie.Title}</p>
        <p className={clsx("text-sm font-medium text-purple-400")}>{movie.Year}</p>
      </div>
    </div >
  )
}

export default Card