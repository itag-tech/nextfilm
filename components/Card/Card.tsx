import { FC } from "react"
import clsx from "clsx"
import Image from "next/image"
import { getPoster } from "../../utils/movieUtils"
import { Movie } from "../../models/Movie"

type CardProps = {
  movie: Movie
}

const Card: FC<CardProps> = ({ movie }) => {
  return (
    <div
      className={clsx("flex flex-col justify-center items-center bg-white",
        "bg-opacity-50 rounded-lg shadow-md truncate w-full p-6")}
    >
      <Image
        src={getPoster(movie)}
        alt={movie.Title}
        width={200}
        height={200}
        className={clsx("h-32 w-32 object-cover shadow-lg")}
      />
      <div className={clsx("flex flex-col justify-center items-center")}>
        <p className={clsx("text-sm md:text-lg font-medium truncate max-w-xxs xl:max-w-xs text-purple-700 m-4 px-6")}>{movie.Title}</p>
        <p className={clsx("text-sm font-medium text-purple-400")}>{movie.Year}</p>
      </div>
    </div >
  )
}

export default Card