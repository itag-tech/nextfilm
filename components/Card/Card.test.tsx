import { describe, expect, it } from "vitest"
import { render } from '@testing-library/react'
import Card from "./Card"

describe("Card", () => {
  // given
  const movie = {
    Title: "The Lord of the Rings: The Fellowship of the Ring",
    Year: "2001",
    imdbID: "tt0120737",
    Type: "movie",
    Poster: "/imagepath.jpg",
    Plot: "A meek Hobbit from the Shire and eight companions set out on a journey to destroy the powerful One Ring and save Middle-earth from the Dark Lord Sauron.",
    Gender: "Action, Adventure, Drama",
  }
  it("should render as usual", () => {

    // when ----------------
    const { container } = render(<Card movie={movie} />)

    // then ----------------
    expect(container).toMatchSnapshot()
  })


  // Test 2
  it("should render a src image", () => {
    // when ----------------
    const { container } = render(<Card movie={movie} />)
    const imageElement = container.querySelector('img')

    // then ----------------
    expect(imageElement).toBeDefined()
    if (imageElement) {
      expect(imageElement.hasAttribute('src')).toBe(true)
      const src = imageElement.getAttribute('src')
      expect(src).contain('/_next/image?url=%2Fimagepath.jpg')
    }
  })
})