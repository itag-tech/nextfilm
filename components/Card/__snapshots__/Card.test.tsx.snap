// Vitest Snapshot v1

exports[`Card > should render as usual 1`] = `
<div>
  <div
    class="flex flex-col justify-center items-center bg-white bg-opacity-50 rounded-lg shadow-md truncate w-full p-6"
  >
    <img
      alt="The Lord of the Rings: The Fellowship of the Ring"
      class="h-32 w-32 object-cover shadow-lg"
      data-nimg="1"
      decoding="async"
      height="200"
      loading="lazy"
      src="/_next/image?url=%2Fimagepath.jpg&w=640&q=75"
      srcset="/_next/image?url=%2Fimagepath.jpg&w=256&q=75 1x, /_next/image?url=%2Fimagepath.jpg&w=640&q=75 2x"
      style="color: transparent;"
      width="200"
    />
    <div
      class="flex flex-col justify-center items-center"
    >
      <p
        class="text-sm md:text-lg font-medium truncate max-w-xxs xl:max-w-xs text-purple-700 m-4 px-6"
      >
        The Lord of the Rings: The Fellowship of the Ring
      </p>
      <p
        class="text-sm font-medium text-purple-400"
      >
        2001
      </p>
    </div>
  </div>
</div>
`;
